library/worldgen/dimension/src/main/java/org/quiltmc/qsl/worldgen/dimension/impl/QuiltDimensionsImpl.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2016, 2017, 2018, 2019 FabricMC
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.worldgen.dimension.impl;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.world.TeleportTarget;
+import org.jetbrains.annotations.ApiStatus;
+import org.quiltmc.qsl.worldgen.dimension.access.EntityAccess;
+
+@ApiStatus.Internal
+public class QuiltDimensionsImpl {
+
+	// Static only-class, no instantiation necessary!
+	private QuiltDimensionsImpl() {
+	}
+
+	@SuppressWarnings("unchecked")
+	public static <E extends Entity> E teleport(Entity entity, ServerWorld destinationWorld, TeleportTarget location) {
+		Preconditions.checkArgument(Thread.currentThread() == ((ServerWorld) entity.getWorld()).getServer().getThread(), "This method may only be called from the main server thread");
```

Comment:
Nit-picky, but this would improve crash reports in some cases!

```suggestion
		Preconditions.checkArgument(Thread.currentThread() == entity.getServer().getThread(), "This method may only be called from the main server thread");
```

Reply:
A similar check is already done from the API method, should that one be replaced by this and moved into here?

Reply:
Edited the review, turns out there's a getServer method in entity already, so no need to use fragile casts.